diff --git a/lib/gro/gro_tcp4.c b/lib/gro/gro_tcp4.c
index aff22178e3..7a84b89dab 100644
--- a/lib/gro/gro_tcp4.c
+++ b/lib/gro/gro_tcp4.c
@@ -221,11 +221,16 @@ gro_tcp4_reassemble(struct rte_mbuf *pkt,
 	hdr_len = pkt->l2_len + pkt->l3_len + pkt->l4_len;
 
 	/*
-	 * Don't process the packet which has FIN, SYN, RST, PSH, URG, ECE
-	 * or CWR set.
+	 * Don't process the packet which has FIN, SYN, RST, URG, ECE or
+	 * CWR set.
+	 * XXX: Openperf only calls the reassemble burst function, so
+	 * packets are never buffered through multiple calls. As a result,
+	 * we can perform GRO even with the PUSH flag. In fact, rejecting
+	 * the PUSH flag here results in out-of-sequence TCP segments.
 	 */
-	if (tcp_hdr->tcp_flags != RTE_TCP_ACK_FLAG)
+	if (tcp_hdr->tcp_flags & ~(RTE_TCP_ACK_FLAG | RTE_TCP_PSH_FLAG)) {
 		return -1;
+	}
 	/*
 	 * Don't process the packet whose payload length is less than or
 	 * equal to 0.
